### PR TITLE
Fix bug where error occurred during job manager launch is not caught as operator error

### PR DESCRIPTION
**Pull request by cw75**
## Describe your changes and why you are making these changes
The original bug was when running with k8s, an extract operator that attempts to read from the local sqlite database will silently fail. The root cause is when error occurs in operator.Launch(), we weren't setting the operator's exec state accordingly.

This PR also fixes a few other smaller issues and comment typos.

@kenxu95 let's sync up tmr on writing a k8s integration test for this.

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: sdk/aqueduct/constants/enums.py [hunk 1]
```diff
@@ -100,6 +100,7 @@ class RelationalDBServices(str, Enum, metaclass=MetaEnum):
 class ExecutionStatus(str, Enum, metaclass=MetaEnum):
     UNKNOWN = "unknown"
     SUCCEEDED = "succeeded"
+    RUNNING = "running"
```
Comment (cw75) [inline sdk/aqueduct/constants/enums.py:103]: We're missing this entry, which is why pydantic was complaining on the SDK side for the preview response.

File: src/golang/lib/job/errors.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ const (
 	// For example, polling a Lambda JobManager will return this error because there is no concept of
 	// fetching a specific job's information from Lambda. The caller must figure out the job status
 	// through other means.
-	Noop = 2
+	Noop = 3
```
Comment (cw75) [inline src/golang/lib/job/errors.go:28]: The original value is duplicated with `JobMissing` above, so I think this is just a bug?

File: src/golang/lib/job/k8s.go [hunk 3]
```diff
@@ -140,7 +140,7 @@ func (j *k8sJobManager) Launch(ctx context.Context, name string, spec Spec) JobE
 
 	containerRepo, err := mapJobTypeToDockerImage(spec, launchGpu)
 	if err != nil {
-		return systemError(err)
+		return userError(err)
```
Comment (cw75) [inline src/golang/lib/job/k8s.go:143]: I think this should be a user error; it catches the k8s <> sqlite compatibility issue

File: src/golang/lib/workflow/operator/base.go [hunk 4]
```diff
@@ -165,7 +165,25 @@ func (bo *baseOperator) launch(ctx context.Context, spec job.Spec) error {
 		}
 	}
 
-	return bo.jobManager.Launch(ctx, spec.JobName(), spec)
+	err := bo.jobManager.Launch(ctx, spec.JobName(), spec)
```
Comment (cw75) [inline src/golang/lib/workflow/operator/base.go:168]: This is the fix to the core bug.

File: src/golang/lib/workflow/operator/base.go [hunk 5]
```diff
@@ -165,7 +165,25 @@ func (bo *baseOperator) launch(ctx context.Context, spec job.Spec) error {
...
+			)
+		} else {
+			log.Errorf("Unexpected job error code %d", err.Code())
+			bo.UpdateExecState(
+				unknownSystemFailureExecState(err, "Job manager Launch failed due to system error"),
+			)
+		}
+	}
+
+	return err
```
Comment (cw75) [inline src/golang/lib/workflow/operator/base.go:186]: This is the fix to the core bug.

File: integration_tests/sdk/aqueduct_tests/integration_test.py [hunk 6]
```diff
@@ -46,3 +50,11 @@ def test_invalid_connect_integration(client):
     # Invalid config raises a pydantic error.
     with pytest.raises(ValidationError):
         client.connect_integration("New Integration", "SQLite", {})
+
+
+@pytest.mark.enable_only_for_engine_type(ServiceType.K8S)
+def test_sqlite_with_k8s(data_integration, engine):
+    """Tests that running an extract operator that reads data from a SQLite database using k8s should fail."""
+    global_config({"engine": engine})
+    with pytest.raises(AqueductError):
```
Comment (kenxu95) [inline integration_tests/sdk/aqueduct_tests/integration_test.py:59]: can you use the "match=<err msg>" argument here to make it clear what the expected error message is?

File: src/golang/lib/workflow/operator/base.go [hunk 7]
```diff
@@ -165,7 +165,25 @@ func (bo *baseOperator) launch(ctx context.Context, spec job.Spec) error {
...
+				jobManagerUserFailureExecState(err, "Job manager Launch failed due to user error"),
+			)
+		} else if err.Code() == job.System {
+			bo.UpdateExecState(
+				unknownSystemFailureExecState(err, "Job manager Launch failed due to system error"),
+			)
+		} else {
+			log.Errorf("Unexpected job error code %d", err.Code())
+			bo.UpdateExecState(
+				unknownSystemFailureExecState(err, "Job manager Launch failed due to system error"),
```
[the remark this answers is not in the document]
Comment (cw75) [reply src/golang/lib/workflow/operator/base.go:181]: @kenxu95 but here we are just logging the err message on the server and not returning it to the client side, so I think revealing the detail here is still fine and useful?

[the remark this answers is not in the document]
Comment (cw75) [reply src/golang/lib/workflow/operator/base.go:181]: @kenxu95 for the unexpected case, what additional detail do you want to add? `unknownSystemFailureExecState` is already logging the error message so I was thinking it should already give the detail?

[the remark this answers is not in the document]
Comment (kenxu95) [reply src/golang/lib/workflow/operator/base.go:181]: I meant like adding a little more detail to the `unknownSystemFailureExecState()` message, but it's not a big deal.

**Comment by cw75** [comment]
@kenxu95 addressed comments.

File: src/golang/lib/job/errors.go [hunk 8]
```diff
@@ -10,22 +10,22 @@ type JobErrorCode int
 
 const (
 	// System indicates an unexpected system issue that we cannot recover from.
-	System JobErrorCode = 0
+	System JobErrorCode = iota
 
 	// User error code indicates that the issue was the user's fault, and to surface that message
 	// to the user.
-	User = 1
+	User = iota
```
Comment (kenxu95) [inline src/golang/lib/job/errors.go:17]: Wait you don't need to set the other ones to iota right, it goes:
```
System JobErrorCode = iota
User
JobMissing
Noop
```

File: src/golang/lib/workflow/operator/base.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (kenxu95) [reply src/golang/lib/workflow/operator/base.go:181]: Can we uncapitalize "Launch" though?

**Review by kenxu95 — APPROVED** [review]
Unblocking but a couple last requests.

File: src/golang/lib/job/errors.go (hunk 8 above)
Comment (cw75) [reply src/golang/lib/job/errors.go:17]: got it thx!

**Comment by cw75** [comment]
@kenxu95 I changed to "Job manager's Launch API..."

**Pull request merged**